app/scripts/services/secrets.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("SecretsService", function($q, $filter, DataService){
+
+    var newestSecret;
+
+    var setNewestSecretName = function(secretName) {
+      newestSecret = secretName;
+    };
+
+    var getNewestSecretName = function() {
+       return newestSecret;
+    };
+
+    var loadSecrets = function(projectName, $scope) {
+      var deferred = $q.defer();
+      var secretsByType = {
+        source: [""],
+        image: [""]
```

Comment:
Really these should be empty arrays. The controller should add the empty item if it needs it for the select drop downs.